code/modules/reagents/chemistry/machinery/reagentgrinder.dm

```diff
@@ -462,6 +459,40 @@
 
 		O.reagents.trans_to(beaker, O.reagents.total_volume)
 
+		if(!O.reagents.total_volume)
+			remove_object(O)
+		if(beaker.reagents.holder_full())
+			return
+	// Inedible Plants
+	for(var/obj/item/grown/O in holdingitems)
+		var/list/special_blend = get_special_blend(O)
+		for(var/r_id in special_blend)
+			var/space = beaker.reagents.maximum_volume - beaker.reagents.total_volume
+			var/amount = special_blend[r_id]
+
+			if(amount <= 0)
+				if(amount == 0)
+					if(O.reagents.has_reagent("nutriment"))
+						beaker.reagents.add_reagent(r_id, min(O.reagents.get_reagent_amount("nutriment") * efficiency, space))
+						O.reagents.remove_reagent("nutriment", min(O.reagents.get_reagent_amount("nutriment"), space))
+					if(O.reagents.has_reagent("plantmatter"))
+						beaker.reagents.add_reagent(r_id, min(O.reagents.get_reagent_amount("plantmatter") * efficiency, space))
+						O.reagents.remove_reagent("plantmatter", min(O.reagents.get_reagent_amount("plantmatter"), space))
+				else
+					if(O.reagents.has_reagent("nutriment"))
+						beaker.reagents.add_reagent(r_id, min(round(O.reagents.get_reagent_amount("nutriment") * abs(amount) * efficiency), space))
+						O.reagents.remove_reagent("nutriment", min(O.reagents.get_reagent_amount("nutriment"), space))
+					if(O.reagents.has_reagent("plantmatter"))
+						beaker.reagents.add_reagent(r_id, min(round(O.reagents.get_reagent_amount("plantmatter") * abs(amount) * efficiency), space))
+						O.reagents.remove_reagent("plantmatter", min(O.reagents.get_reagent_amount("plantmatter"), space))
+			else
+				O.reagents.trans_id_to(beaker, r_id, min(amount, space))
```

Comment:
```suggestion
			if(amount == 0)
				if(O.reagents.has_reagent("nutriment"))
					beaker.reagents.add_reagent(r_id, min(O.reagents.get_reagent_amount("nutriment") * efficiency, space))
					O.reagents.remove_reagent("nutriment", min(O.reagents.get_reagent_amount("nutriment"), space))
				if(O.reagents.has_reagent("plantmatter"))
					beaker.reagents.add_reagent(r_id, min(O.reagents.get_reagent_amount("plantmatter") * efficiency, space))
					O.reagents.remove_reagent("plantmatter", min(O.reagents.get_reagent_amount("plantmatter"), space))
			else if(amount < 0)
					if(O.reagents.has_reagent("nutriment"))
						beaker.reagents.add_reagent(r_id, min(round(O.reagents.get_reagent_amount("nutriment") * abs(amount) * efficiency), space))
						O.reagents.remove_reagent("nutriment", min(O.reagents.get_reagent_amount("nutriment"), space))
					if(O.reagents.has_reagent("plantmatter"))
						beaker.reagents.add_reagent(r_id, min(round(O.reagents.get_reagent_amount("plantmatter") * abs(amount) * efficiency), space))
						O.reagents.remove_reagent("plantmatter", min(O.reagents.get_reagent_amount("plantmatter"), space))
			else
				O.reagents.trans_id_to(beaker, r_id, min(amount, space))
```

Reply:
so just removing this bit at the beginning? 
>if(amount == 0)

Reply:
oh nevermind I understand

Reply:
Needs to be done for the edible plants instead

Reply:
done